state-chain/custom-rpc/src/lib.rs

```diff
@@ -1086,6 +1086,35 @@ where
 	) -> Self {
 		Self { rpc_backend: CustomRpcBackend::new(client, backend, executor) }
 	}
+
+	pub fn into_block_update<Data>(
+		&self,
+		data: RpcResult<Data>,
+		at: Option<Hash>,
+	) -> RpcResult<BlockUpdate<Data>> {
+		let block_hash = self.rpc_backend.unwrap_or_best(at);
+		let block_number = self
+			.rpc_backend
+			.client
+			.number(block_hash)
+			.map_err(|e| call_error(e, CfErrorCode::OtherError))?
+			.ok_or(internal_error(format!("Could not fetch block number for block {:?}", at)))?;
```

Comment:
It would be good to understand the circumstances that can cause this to fail: what if the rpc is called on a block that is no longer in the block cache? (Like on a light rpc node).